centreon/www/class/centreonGraph.class.php

```diff
@@ -1662,53 +1676,79 @@ private function manageVMetric($vId, $vName, $indexId): void
             if (!isset($this->mlist["v" . $lVmetric["metric_id"]])) {
                 if (is_null($vId)) {
                     $lVmetric["need"] = 1; /* 1 : Need this virtual metric : Hidden */
                 }
                 /*
                  * Find Host/Service For this metric_id
                  */
-                $l_poqy = $this->DBC->query(
-                    "SELECT host_id, service_id FROM index_data WHERE id = '" . $lVmetric["index_id"] . "'"
+                $selectIndexDataStatement = $this->DBC->prepareQuery(
+                    <<<SQL
+                    SELECT host_id, service_id FROM index_data WHERE id = :indexId
+                    SQL
                 );
-                $l_indd = $l_poqy->fetch();
-                $l_poqy->closeCursor();
+                $this->DBC->executePreparedQuery(
+                    $selectIndexDataStatement,
+                    [
+                        'indexId' => $lVmetric["index_id"]
+                    ]
+                );
+                $l_indd = $this->DBC->fetch($selectIndexDataStatement);
+                $this->DBC->closeQuery($selectIndexDataStatement);
                 /* Check for real or virtual metric(s) in the RPN function */
                 $l_mlist = preg_split("/\,/", $lVmetric["rpn_function"]);
                 foreach ($l_mlist as $l_mnane) {
                     /*
                      * Check for a real metric
                      */
-                    $l_poqy = $this->DBC->query(
-                        "SELECT host_id, service_id, metric_id, metric_name, unit_name,
+                    $checkRealMetricStatement = $this->DBC->prepareQuery(
+                        <<<SQL
+                        SELECT host_id, service_id, metric_id, metric_name, unit_name,
                             replace(format(warn,9),',','') warn, replace(format(crit,9),',','') crit
                             FROM metrics AS m, index_data as i
-                            WHERE index_id = id AND index_id = '" . $lVmetric["index_id"] . "'
-                                AND metric_name = '" . $l_mnane . "'"
+                            WHERE index_id = id AND index_id = :indexId
+                                AND metric_name = :vMetricName
+                        SQL
+                    );
+                    $this->DBC->executePreparedQuery(
+                        $checkRealMetricStatement,
+                        [
+                            'indexId' => $lVmetric["index_id"],
+                            'vMetricName' => $l_mnane
+                        ]
                     );
-                    if ($l_poqy->rowCount() == 1) {
+                    if ($checkRealMetricStatement->rowCount() === 1) {
                         /* Find a real metric in the RPN function */
-                        $l_rmetric = $l_poqy->fetchrow();
-                        $l_poqy->closeCursor();
+                        $l_rmetric = $this->DB->fetch($checkRealMetricStatement);
```

Comment:
Catch CentreonDbException and log the error with CentreonLog with business informations and a context please

Reply:
Seen together, you don't want to change your code this time to merge it quickly. 
But the next time, can you catch the exception and use the logger please to improve our error handling ? Thanks.